### PR TITLE
add causingPlayer to McMMOReplaceVanillaTreasureEvent and update Fish Event to use it

### PR DESCRIPTION
**Pull request by bobcat4848**
Improvement to the McMMOReplaceVanillaTreasureEvent to include a causing player.

### CODE REVIEW

File: src/main/java/com/gmail/nossr50/events/McMMOReplaceVanillaTreasureEvent.java [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
 import org.bukkit.event.Event;
 import org.bukkit.event.HandlerList;
 import org.bukkit.inventory.ItemStack;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public class McMMOReplaceVanillaTreasureEvent extends Event {
     private @NotNull ItemStack replacementItemStack;
     private final @NotNull Item originalItem;
+    private @Nullable Player causingPlayer;
```
Comment (nossr50) [inline src/main/java/com/gmail/nossr50/events/McMMOReplaceVanillaTreasureEvent.java:14]: make final

**Review by nossr50 — CHANGES_REQUESTED** [review]
Looks great, one small change needed.

File: src/main/java/com/gmail/nossr50/events/McMMOReplaceVanillaTreasureEvent.java (hunk 1 above)
Comment (bobcat4848) [reply src/main/java/com/gmail/nossr50/events/McMMOReplaceVanillaTreasureEvent.java]: Done!

**Comment by nossr50** [comment]
LGTM 👍

**Pull request merged**